pkg/cloudprovider/providers/azure/azure.go

```diff
@@ -118,6 +121,12 @@ type Config struct {
 	CloudProviderBackoffDuration int `json:"cloudProviderBackoffDuration" yaml:"cloudProviderBackoffDuration"`
 	// Backoff jitter
 	CloudProviderBackoffJitter float64 `json:"cloudProviderBackoffJitter" yaml:"cloudProviderBackoffJitter"`
+	// Backoff mode, options are minimum and default.
+	// * default means two-layer backoff retrying, one in the cloud provider and the other in the Azure SDK.
+	// * minimum means only backoff in the Azure SDK is used. In such mode, CloudProviderBackoffDuration and
+	//   CloudProviderBackoffJitter are omitted.
+	// "default" will be used if not specified.
```

Comment:
I would suggest to use sdk retry flavor by default, we have 3 months to test this retry change, and if we don't change the original retry mechanism, I doubt whether aks or acs-engine will use sdk retry flavor, the new retry machanism will never be used.

Reply:
@brendandburns @khenidak What do you think about the default backoff mode?

Reply:
I don't think we should maintain two modes. Do you see a usecase for maintaining the old mode?

Reply:
per @brendandburns suggestion [here](https://github.com/kubernetes/kubernetes/pull/70866#issuecomment-438447463). 

> I'm a little bit worried about taking out the manual configuration knobs and deferring entirely to the SDK.

> I think that we should both maintain the existing behavior, and implement the new one, and switch between them based on a flag.

> Since this is a big change, we need to maintain backward/rollback as an option for clusters without reverting the Kubernetes version.